common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -87,15 +91,26 @@ public static void submitSilent(@NonNull final BaseCommand command) {
                 "Beginning execution of silent command."
         );
 
-        if (sExecutingCommands.contains(command)) {
-            new Handler(Looper.getMainLooper()).post(new Runnable() {
-                @Override
-                public void run() {
-                    command.getCallback().onError(new ClientException(ClientException.DUPLICATE_COMMAND, "The same command was already received and is being processed."));
-                }
-            });
-        } else if (command.isEligibleForCaching()) {
-            sExecutingCommands.add(command);
+        final Handler handler = new Handler(Looper.getMainLooper());
+        ResultFuture<CommandResult> future = sExecutingCommandMap.get(command);
+
+        if (null != future) {
+            future.whenComplete(getCommandResultConsumer(command, handler));
+            return;
+        } else {
+            final ResultFuture<CommandResult> resultFuture = new ResultFuture<>();
+            final ResultFuture<CommandResult> putValue = sExecutingCommandMap.putIfAbsent(command, resultFuture);
+
+            if (null == putValue) {
+                // our value was inserted.
+                resultFuture.whenComplete(getCommandResultConsumer(command, handler));
+            } else {
+                // Our value was not inserted, grab the one that was and hang a new listener off it
+                putValue.whenComplete(getCommandResultConsumer(command, handler));
+                return;
+            }
+
+            sExecutingCommandMap.put(command, resultFuture);
```

Comment:
Don't think you need this.

Reply:
Already removed, just didn't push the change -- it's in b908342ba087ef5bccca5e7eb486ea49093da911